File: src/main/java/ru/tvgtu/dutystatistics/web/dto/DutyObjectRouteDTO.java
package ru.tvgtu.dutystatistics.web.dto;

import lombok.Data;
import ru.tvgtu.dutystatistics.model.Route;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
public class DutyObjectRouteDTO {

    private UUID id;

    /**
     * Длина маршрута
     */
    private Long length;

    /**
     * Наработка полного технического ресурса двигателя, ч
     */
    private Long operatingFullResource;

    /**
     * Дата начала маршрута
     */
    private LocalDateTime startDate;

    /**
     * Дата окончания маршрута
     */
    private LocalDateTime endDate;

    public DutyObjectRouteDTO(Route route) {
        this.id = route.getId();
        this.length = route.getLength();
        this.operatingFullResource = route.getOperatingFullResource();
        this.startDate = route.getStartDate();
        this.endDate = route.getEndDate();
    }
}

File: web/src/models/DutyObjectRouteDTO.ts
import moment from 'moment';

export class DutyObjectRouteDTO {
    public id!: string;
    /**
     * Длина маршрута, км
     */
    public length: number | null = null;
    public operatingFullResource: number | null = null;
    public startDate: Date = new Date();
    public endDate: Date = new Date();

    public get formatDate(): string {
        return `от ${moment(this.startDate).format('DD.MM.YYYY')}`;
    }
}
